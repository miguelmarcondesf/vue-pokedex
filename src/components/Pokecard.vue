<template>
  <v-card>
    <v-container fluid grid-list-md>
      <v-layout row wrap>
        <v-flex>
          <v-card color="grey" class="white--text">
            <v-layout row>
              <v-flex aling-self-end>
                <v-card-title primary-title>
                  <div>
                    <h1 class="mr-4"># {{ pokeId }}</h1>
                  </div>
                  <div>
                    <div class="headline text-capitalize">{{ pokeName }}</div>
                    <div>
                      <v-chip
                        v-for="(type, index) in pokeTypes"
                        :key="index"
                        color="primary"
                        text-color="white"
                      >{{type.type.name}}</v-chip>
                    </div>
                  </div>
                </v-card-title>
              </v-flex>
              <v-flex xs5>
                <v-img :src="pokeImage" height="125px" contain></v-img>
              </v-flex>
            </v-layout>
            <v-divider light></v-divider>
          </v-card>
        </v-flex>
      </v-layout>
    </v-container>
  </v-card>
</template>

<script>
import { mapGetters } from "vuex";

export default {
  props: {
    id: 0,
    name: "",
    types: [],
    image: ""
  },
  
  data () {
    return {
      pokeId: 0,
      pokeName: "",
      pokeTypes: [],
      pokeImage: ""
    }
  },

  methods: {
    getInfo() {
      this.$http.get(`pokemon/${this.name}`).then(res => {
        const data = res.data;
        this.pokeTypes = data.types;
        this.pokeImage = data.sprites.front_default;
        this.pokeId = data.id;
        this.pokeName = data.name
      });
    }
  },

  mounted() {
    this.$nextTick(() => {
      this.getInfo()
    })
  }

};
</script>

<style>
</style>
