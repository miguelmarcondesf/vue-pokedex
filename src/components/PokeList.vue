<template>
  <v-item-group>
    <input type="text" placeholder="search" v-model="search">

    <v-container grid-list-md>
      <v-layout wrap>
        <v-flex
          v-for="(pokemon, index) in filteredPokemon"
          :key="index"
          xs12
          sm6
          md4
        >
          <v-item @click="goToPokemon(pokemon.name)">
            <Pokecard :name="pokemon.name" />
          </v-item>
        </v-flex>
      </v-layout>
    </v-container>
  </v-item-group>
</template>

<script>
import Pokecard from './Pokecard'

export default {
  data () {
    return {
      search: ''
    }
  },

  components: {
    Pokecard
  },

  props: {
    pokeList: {
      type: Array,
      default () {
        return []
      }
    }
  },

  computed: {
    filteredPokemon () {
      console.log(this.search)

      return this.pokeList.filter((pokemon) => {
        return pokemon.name.match(this.search)
      })
    }
  }
}
</script>

<style>
  v-card {
    margin: 8px;
  }
</style>
