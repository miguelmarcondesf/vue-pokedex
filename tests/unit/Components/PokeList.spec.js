import { shallowMount } from '@vue/test-utils'
import PokeList from '@/components/PokeList'
import Pokecard from '@/components/Pokecard'
import Vue from 'vue'
import Vuetify from 'vuetify'

Vue.use(Vuetify)

describe('PokeList.vue', () => {
  let pokeList

  beforeEach(() => {
    pokeList = [
      {
        'name': 'bulbasaur',
        'url': 'https://pokeapi.co/api/v2/pokemon/1/'
      },
      {
        'name': 'ivysaur',
        'url': 'https://pokeapi.co/api/v2/pokemon/2/'
      },
      {
        'name': 'venusaur',
        'url': 'https://pokeapi.co/api/v2/pokemon/3/'
      }
    ]
  })

  it('is a Vue instance', () => {
    const wrapper = shallowMount(PokeList, {
      propsData: {
        pokeList: pokeList
      }
    })
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  it('should render content correctly', () => {
    const wrapper = shallowMount(PokeList, {
      propsData: {
        pokeList: pokeList
      }
    })

    expect(wrapper.vm.$el).toMatchSnapshot()
  })

  it('should render 3 pokecards', () => {
    const wrapper = shallowMount(PokeList, {
      propsData: {
        pokeList: pokeList
      }
    })

    expect(wrapper.findAll(Pokecard).length).toBe(2)
  })
})
