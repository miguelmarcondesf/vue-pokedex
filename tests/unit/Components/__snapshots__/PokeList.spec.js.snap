// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokeList.vue should render content correctly 1`] = `
<v-item-group-stub
  activeclass="v-item--active"
>
  <input
    placeholder="search"
    type="text"
  />
   
  <v-container-stub
    grid-list-md=""
    tag="div"
  >
    <v-layout-stub
      tag="div"
      wrap=""
    >
      <v-flex-stub
        md4=""
        sm6=""
        tag="div"
        xs12=""
      >
        <v-item-stub>
          <pokecard-stub
            name="bulbasaur"
          />
        </v-item-stub>
      </v-flex-stub>
      <v-flex-stub
        md4=""
        sm6=""
        tag="div"
        xs12=""
      >
        <v-item-stub>
          <pokecard-stub
            name="ivysaur"
          />
        </v-item-stub>
      </v-flex-stub>
      <v-flex-stub
        md4=""
        sm6=""
        tag="div"
        xs12=""
      >
        <v-item-stub>
          <pokecard-stub
            name="venusaur"
          />
        </v-item-stub>
      </v-flex-stub>
    </v-layout-stub>
  </v-container-stub>
</v-item-group-stub>
`;
